src/generators/operationGenerator.ts

```diff
@@ -109,6 +109,26 @@ function generateOperation(
   );
 }
 
+function writeGetOperationOptions(operationGroupClass: ClassDeclaration) {
+  operationGroupClass.addMethod({
+    scope: Scope.Private,
+    name: "getOperationOptions<TOptions extends coreHttp.OperationOptions>",
+    parameters: [
+      { name: "options", type: "TOptions | undefined" },
+      { name: "finalStateVia", type: "string", hasQuestionToken: true }
+    ],
+    returnType: `coreHttp.RequestOptionsBase`,
+    statements: `
+    const operationOptions: coreHttp.OperationOptions = options || {};
+    operationOptions.requestOptions = {
+      ...operationOptions.requestOptions,
+      shouldDeserialize: shouldDeserializeLRO(finalStateVia),
+    };
+    return coreHttp.operationOptionsToRequestOptionsBase(operationOptions);
```

Comment:
I think I missed converting to RequestOptionsBase when I made the switch from RequestOptionsBase to OperationOptions.

Since RequestOptionsBase has an indexer it wouldn't complain about us not converting it, but I think SendOperation  doesn't understand how to flatten OperationOptions and would silently fail to apply the right options